### ISSUE TITLE
bootstrap-ansible.sh only installs ceph roles, misses keepalived

### ISSUE THREAD

**Issue by shane-c** [rcbops/rpc-openstack#971]
##### What version (commit) do you encounter this issue on?

r12.0.0

##### What type of issue is this (bug, feature, documentation, regression)?

bug

##### What is the expected behavior?

bootstrap-ansible.sh should install all external roles, ceph and keepalived

##### What is actually happening?

bootstrap-ansible.sh only installs the ceph roles and misses the keepalived role

##### How can we reproduce this issue?

Run ./openstack-ansible/scripts/bootstrap-ansible.sh
Check /etc/ansible/roles

##### How can we verify your issue is addressed?

Running haproxy-install.yml doesn't run into the following error like it does now, both ceph and keepalived roles are installed

ERROR: cannot find role in /opt/rpc-openstack/openstack-ansible/playbooks/roles/keepalived or /opt/rpc-openstack/openstack-ansible/playbooks/keepalived or /etc/ansible/roles/keepalived

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?

Combine the OSA and rpc-openstack ansible-role-requirements.yml files or make sure the bootstrap-ansible.sh script installs from both files

root@infra01:/opt/rpc-openstack# find . -name ansible-role-requirements.yml
./openstack-ansible/ansible-role-requirements.yml
./ansible-role-requirements.yml

##### What is your evaluation on the importance of this issue, and why?

Low, easily worked around

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.1
- [ ] liberty-12.0

**Comment by odyssey4me** [comment]
This is likely because the use of haproxy/keepalived is, as yet, unsupported from an RPC standpoint as far as I know. I think it's on the roadmap, but hasn't become a supported configuration yet from a development team standpoint.

**Comment by shane-c** [comment]
That might be, but pre-Liberty the haproxy-install.yml playbook worked just fine in rpc-openstack. With Liberty and the move to ansible-role-requirements.yml files for the external roles, it doesn't. IMO it's still something that should be fixed, since before Liberty this worked. 

With my test I didn't even want to use keepalived, I was testing with one haproxy host, but currently that won't even finish unless you manaully cd /opt/rpc-openstack/openstack-ansible/; ansible-galaxy install --force --role-file=ansible-role-requirements.yml to get the keepalived role.

**Comment by odyssey4me** [comment]
It does, in fact, work exactly as it did before as long as you only have one haproxy host. The keepalived role is only used when more than one haproxy host is configured. This matches the exact same functionality as it always has.

**Comment by shane-c** [comment]
That's not correct. I only had one haproxy host in my openstack_user_config.yml file and it still errored out for the keepalived role not being present.

**Comment by txgunslinger77** [comment]
I just ran into this error as well on a Liberty 12.1.0 deployment that is not even using or has ever used haproxy:
root@XXXXXX-ocp:/opt/rpc-openstack/openstack-ansible/playbooks# openstack-ansible setup-infrastructure.yml
Variable files: "-e @/etc/openstack_deploy/user_extras_secrets.yml -e @/etc/openstack_deploy/user_extras_variables.yml -e @/etc/openstack_deploy/user_secrets.yml -e @/etc/openstack_deploy/user_variables.yml "
ERROR: cannot find role in /opt/rpc-openstack/openstack-ansible/playbooks/roles/keepalived or /opt/rpc-openstack/openstack-ansible/playbooks/keepalived or /etc/ansible/roles/keepalived

**Comment by shane-c** [comment]
Also came across this with the opposite scenario. keepalived role was installed and ceph roles were not. Even though I do not have any ceph hosts in this environment, it's still looking for the ceph roles. Ideally for this entire scenario, the playbooks would only need the required roles based on what you have in the openstack_user_config.yml file. 

root@infra01:/opt/rpc-openstack/rpcd/playbooks# openstack-ansible site.yml
Variable files: "-e @/etc/openstack_deploy/user_extras_secrets.yml -e @/etc/openstack_deploy/user_extras_variables.yml -e @/etc/openstack_deploy/user_secrets.yml -e @/etc/openstack_deploy/user_variables.yml "

ERROR: cannot find role in /opt/rpc-openstack/rpcd/playbooks/roles/ceph.ceph-common or /opt/rpc-openstack/rpcd/playbooks/ceph.ceph-common or /opt/rpc-openstack/openstack-ansible/playbooks/roles/ceph.ceph-common or /etc/ansible/roles/ceph.ceph-common

root@infra01:/opt/rpc-openstack# ls -la /etc/ansible/roles/
total 20
drwxr-xr-x  5 root root 4096 Jun  1 14:43 .
drwxr-xr-x  3 root root 4096 Jun  1 14:43 ..
drwxr-xr-x  9 root root 4096 Jun  1 14:43 keepalived
drwxr-xr-x 12 root root 4096 Jun  1 14:43 openstack-ansible-security
drwxr-xr-x  9 root root 4096 Jun  1 14:43 sshd
